docs/contributing/Android-API-And-Related-Versions.md

```diff
@@ -0,0 +1,115 @@
+# Android Api and Related versions
+
+Picking the full suite of android versions can be difficult if you are not a full time android developer. 
+This document is for contributors to understand what versions of common android tooling should be used in the flutter/flutter codebase. 
+
+## Overview
+
+**The most important thing is for any test, project, or app to have compatible versions. **
+
+If versions need to be different then a comment explaining why and what is being evaluated should be included. 
+All chosen versions should pass the minimum versions checks defined in [DependencyVersionChecker](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/gradle/src/main/kotlin/DependencyVersionChecker.kt) 
+
+If the versions chosen are not known by [gradle_utils.dart](https://github.com/flutter/flutter/blob/a16c447abcb695b4ca907d59e66dc87f4f7178d3/packages/flutter_tools/lib/src/android/gradle_utils.dart#L63) then update gradle_utils.dart. 
+
+## Specifics
+
+- "compileSdk" version should be `flutter.compileSdkVersion` or if a pure android project the highest number that has proven stable and is available in CIPD. 
+- "targetSdk" takes human level effort to update. This is by design of the AGP team. The version should be `flutter.targetSdkVersion` or the highest number that has proven stable and is available in cipd. This number may lag behind compile sdk because of the effort to update.
+- "AGP" version should be the version set in flutter templates or newer. If the version is intentionally different there should be a comment explaining why.
+- "gradle" version should be the version set in flutter templates or newere. If the version is intentionally different there should be a comment explaining why.
+- "kotlin" version should be the version set in flutter templates or newere. If the version is intentionally different there should be a comment explaining why.
+
+### compileSdk 
+
+Must be higher then or equal to `targetSdk`. 
```

Comment:
Here and below: Make these bullet points. As written it looks like one long sentence when markdowned:

<img width="920" alt="Screenshot 2025-02-18 at 5 14 53 PM" src="https://github.com/user-attachments/assets/e741db2b-9419-43e1-83e4-1eaaef215ee8" />

Reply:
done